server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1739,36 +1740,39 @@ public void getRepositoryData(ActionListener<RepositoryData> listener) {
         // master-eligible or not.
         assert clusterService.localNode().isMasterNode() : "should only load repository data on master nodes";
 
-        if (lifecycle.started() == false) {
-            listener.onFailure(notStartedException());
-            return;
-        }
+        while (true) {
```

Comment:
NB previously we were retrying via recursion, probably not a problem but still seems unnecessary.